pyfar/classes/coordinates.py

```diff
@@ -606,10 +914,264 @@ def csize(self):
+
+    @spherical_elevation.setter
+    def spherical_elevation(self, value):
+        value[..., 1] = _check_range_angle(value[..., 1], -np.pi/2, np.pi/2)
+        x, y, z = sph2cart(
+            value[..., 0], np.pi / 2 - value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_colatitude(self):
+        """returns :py:func:`azimuth`, :py:func:`colatitude`,
+        :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        azimuth, colatitude, radius = cart2sph(self.x, self.y, self.z)
+        return np.atleast_2d(np.moveaxis(
+            np.array([azimuth, colatitude, radius]), 0, -1))
+
+    @spherical_colatitude.setter
+    def spherical_colatitude(self, value):
+        value[..., 1] = _check_range_angle(value[..., 1], 0, np.pi)
+        x, y, z = sph2cart(value[..., 0], value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_side(self):
+        """returns :py:func:`lateral`, :py:func:`polar`, :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        polar, lateral, radius = cart2sph(self.x, self.z, -self.y)
+        lateral = lateral - np.pi / 2
+        polar = np.mod(polar + np.pi / 2, 2 * np.pi) - np.pi / 2
+        return np.atleast_2d(np.moveaxis(
+            np.array([lateral, polar, radius]), 0, -1))
+
+    @spherical_side.setter
+    def spherical_side(self, value):
+        value[..., 0] = _check_range_angle(value[..., 0], -np.pi/2, np.pi/2)
+        x, z, y = sph2cart(
+            value[..., 1], np.pi / 2 - value[..., 0], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def spherical_front(self):
+        """returns :py:func:`phi`, :py:func:`theta`, :py:func:`radius`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+
+        phi, theta, radius = cart2sph(self.y, self.z, self.x)
+        return np.atleast_2d(np.moveaxis(
+            np.array([phi, theta, radius]), 0, -1))
+
+    @spherical_front.setter
+    def spherical_front(self, value):
+        value[..., 1] = _check_range_angle(value[..., 1], 0, np.pi)
+        y, z, x = sph2cart(value[..., 0], value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def cylindrical(self):
+        """returns :py:func:`azimuth`, :py:func:`z`, :py:func:`rho`. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information."""
+        azimuth, z, rho = cart2cyl(self.x, self.y, self.z)
+        return np.atleast_2d(np.moveaxis(
+            np.array([azimuth, z, rho]), 0, -1))
+
+    @cylindrical.setter
+    def cylindrical(self, value):
+        x, y, z = cyl2cart(value[..., 0], value[..., 1], value[..., 2])
+        self._set_points(x, y, z)
+
+    @property
+    def x(self):
```

Comment:
I would suggest to include something like 'Set or get the...' in all docstring of coordinates (see above)